test/unit-test/cellular_at_core_utest.c

```diff
@@ -0,0 +1,932 @@
+
+/**
+ * @brief Cellular sample digit string.
+ */
+#define CELLULAR_SAMPLE_DIGIT_STRING                       "1234"
+
+/**
+ * @brief Cellular sample digit string with NULL.
+ */
+#define CELLULAR_SAMPLE_DIGIT_STRING_WITH_NULL             "1234\"\0\"123"
+
+/**
+ * @brief Cellular sample success code string.
+ */
+#define CELLULAR_SAMPLE_SUCCESS_CODE_STRING                "TEST1:SUCCESS FOR TRANSMISSION"
+
+/**
+ * @brief Cellular sample error code string.
+ */
+#define CELLULAR_SAMPLE_ERROR_CODE_STRING                  "TEST2:ERROR FOR TRANSMISSION"
+
+/**
+ * @brief Cellular sample strtol error case string.
+ */
+#define CELLULAR_SAMPLE_STRTOL_ERROR_CASE_STRING           "20210487 This is test"
+
+/**
+ * @brief Cellular sample strtol happy case string.
+ */
+#define CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_STRING           "20210487"
+
+/**
+ * @brief Cellular sample strtol happy case number.
+ */
+#define CELLULAR_SAMPLE_STRTOL_HAPPY_CASE_NUMBER           ( 20210487 )
+
+/**
+ * @brief Cellular sample prefix string wron input.
+ */
+#define CELLULAR_SAMPLE_PREFIX_STRING_STAR_FIRST_INPUT     "*CPIN:READY"
+
+/* ============================   UNITY FIXTURES ============================ */
+
+/* Called before each test method. */
+void setUp()
+{
+    
+}
+
+/* Called after each test method. */
+void tearDown()
+{
+}
+
+/* Called at the beginning of the whole suite. */
+void suiteSetUp()
+{
+}
+
+/* Called at the end of the whole suite. */
+int suiteTearDown( int numFailures )
+{
+    return numFailures;
+}
+
+/* ========================================================================== */
+
+/* static function declaration */
+
+
+/* ========================================================================== */
+
+/**
+ * @brief Test that any NULL parameter causes Cellular_ATRemovePrefix to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemovePrefix_Invalid_Param( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+
+    /* Check that CELLULAR_BAD_PARAMETER is returned. */
+    cellularStatus = Cellular_ATRemovePrefix( NULL );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_BAD_PARAMETER, cellularStatus );
+}
+
+/**
+ * @brief Test happy path for Cellular_ATRemovePrefix to return CELLULAR_AT_SUCCESS and got expected results.
+ */
+void test_Cellular_ATRemovePrefix_Happy_Path( void )
+{
+    CellularATError_t cellularStatus = CELLULAR_AT_SUCCESS;
+    char * pString = CELLULAR_SAMPLE_PREFIX_STRING_INPUT;
+
+    /* Check that CELLULAR_AT_SUCCESS is returned . */
+    cellularStatus = Cellular_ATRemovePrefix( &pString );
+    TEST_ASSERT_EQUAL( CELLULAR_AT_SUCCESS, cellularStatus );
+
+    /* Test the result to check its functionality. */
+    TEST_ASSERT_EQUAL_STRING( CELLULAR_SAMPLE_PREFIX_STRING_OUTPUT, pString );
+}
+
+/**
+ * @brief Test the error path for Cellular_ATRemovePrefix to return CELLULAR_AT_BAD_PARAMETER.
+ */
+void test_Cellular_ATRemovePrefix_Error_Path( void )

```

Comment:
I'd break it in 3 tests:
* test_Cellular_ATRemovePrefix_Empty_String
* test_Cellular_ATRemovePrefix_Too_Large_String
* test_Cellular_ATRemovePrefix_Invalid_String

Reply:
sure.